contracts/governance/compatibility/GovernorCompatibilityBravo.sol

```diff
@@ -0,0 +1,267 @@
+        uint256 proposalId,
+        address proposer,
+        address[] memory targets,
+        uint256[] memory values,
+        string[] memory signatures,
+        bytes[] memory calldatas,
+        string memory description
+    ) private {
+        ProposalDetails storage details = _proposalDetails[proposalId];
+
+        details.proposer = proposer;
+        details.targets = targets;
+        details.values = values;
+        details.signatures = signatures;
+        details.calldatas = calldatas;
+        details.descriptionHash = keccak256(bytes(description));
+    }
+
+    // ==================================================== Views =====================================================
+    /**
+     * @dev See {IGovernorCompatibilityBravo-proposals}.
+     */
+    function proposals(uint256 proposalId) public view virtual override returns (Proposal memory) {
+        Proposal memory result;
+        result.id = proposalId;
+        result.eta = proposalEta(proposalId);
+
+        ProposalDetails storage details = _proposalDetails[proposalId];
+        result.proposer = details.proposer;
+        result.targets = details.targets;
+        result.values = details.values;
+        result.signatures = details.signatures;
+        result.calldatas = details.calldatas;
+        result.forVotes = details.forVotes;
+        result.againstVotes = details.againstVotes;
+        result.abstainVotes = details.abstainVotes;
+
+        ProposalCore memory core = _getProposal(proposalId);
+        result.startBlock = core.voteStart.getDeadline();
+        result.endBlock = core.voteEnd.getDeadline();
+
+        ProposalState status = state(proposalId);
+        result.canceled = status == ProposalState.Canceled;
+        result.executed = status == ProposalState.Executed;
+        return result;
+    }
+
+    /**
+     * @dev See {IGovernorCompatibilityBravo-getActions}.
+     */
+    function getActions(uint256 proposalId)
+        public
+        view
+        virtual
+        override
+        returns (
+            address[] memory targets,
+            uint256[] memory values,
+            string[] memory signatures,
+            bytes[] memory calldatas
+        )
+    {
+        ProposalDetails storage details = _proposalDetails[proposalId];
+        return (details.targets, details.values, details.signatures, details.calldatas);
+    }
+
+    /**
+     * @dev See {IGovernorCompatibilityBravo-getReceipt}.
+     */
+    function getReceipt(uint256 proposalId, address voter) public view virtual override returns (Receipt memory) {
+        return _proposalDetails[proposalId].receipts[voter];
+    }
+
+    /**
+     * @dev See {IGovernorCompatibilityBravo-quorumVotes}.
+     */
+    function quorumVotes() public view virtual override returns (uint256) {
+        return quorum(block.number);
+    }
+
+    // ==================================================== Voting ====================================================
```

Comment:
`VotingSimple`'s `_pushVote` does not store everything needed by the compatibility layer.

If we were inheriting `VotingSimple`, then we would need to override `_pushVote` to store additional information. In order to have the receipt structure (from compound) already available, I want to store everything like compound does. considering that the voting it the most gas-critical part, we would have to remove the storage that is done by `VotingSimple`'s `_pushVote` ... and very quickly we are overriding everything in the `VotingSimple` module (since storage is somewhere else).

It felt way simpler to not inherit it, and just include a voting module in the compatibility layer.